### PR TITLE
Add guide on how to release ArcticDB to PyPi and conda-forge

### PR DESCRIPTION
**Pull request by mehertz**
Adds a basic release guide.

### CODE REVIEW

File: docs/mkdocs/docs/technical/releasing.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Release Tooling
+
+This document details the release process for ArcticDB. ArcticDB is released onto [PyPi](https://pypi.org/project/arcticdb/) and [Conda](https://anaconda.org/conda-forge/arcticdb).
```
Comment (jjerphan) [inline docs/mkdocs/docs/technical/releasing.md:3]:
```suggestion
This document details the release process for ArcticDB. ArcticDB is released onto [PyPi](https://pypi.org/project/arcticdb/) and [conda-forge](https://anaconda.org/conda-forge/arcticdb).
```

Comment (qc00) [inline docs/mkdocs/docs/technical/releasing.md:3]: Please break long lines in markdown

File: docs/mkdocs/docs/technical/releasing.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+1. The version, pointing to the tag created in step 1. 
+2. The sha of the source tarball
+3. Dependencies (if they need to be changed)
+
+Don't forget to re-render!
+
+## 4. Release to PyPi
+
+After building, GitHub Actions job you kicked off in step 2 after comitting the tag will be waiting on approval to deploy to PyPi. Find the job and click approve to deploy.
```
Comment (jjerphan) [inline docs/mkdocs/docs/technical/releasing.md:51]:
```suggestion
After building, GitHub Actions job you kicked off in step 2 after committing
the tag will be waiting on approval to deploy to PyPi.
Find the job and click approve to deploy.
```

File: docs/mkdocs/docs/technical/releasing.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+git commit -m 'Manually update version'
+git tag <NEWLY CREATED TAG> -f
+git push origin <NEWLY CREATED TAG> -f
+```
+
+The result should look similar to [this commit](https://github.com/man-group/ArcticDB/commit/c90a21a611b5c6ec2ef4b049981ac5c2ccb8ad08).
+
+The [build will now be running for the tag.](https://github.com/man-group/ArcticDB/actions/workflows/build.yml)
+
+## 3. Update conda-forge recipe
```
Comment (qc00) [inline docs/mkdocs/docs/technical/releasing.md:37]: We should add an issue to automate this.

**Review by jjerphan — APPROVED** [review]
Thank you for documenting this, @mehertz.

**Review by jamesmunro — APPROVED** [review]

File: docs/mkdocs/docs/technical/releasing.md (hunk 1 above)
Comment (mehertz) [reply docs/mkdocs/docs/technical/releasing.md]: All long lines split. Thanks for the reminder.

File: docs/mkdocs/docs/technical/releasing.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+Create a new PR to update the Conda recipe. [Here's an example.](https://github.com/conda-forge/arcticdb-feedstock/pull/10)
```
[the remark this answers is not in the document]
Comment (mehertz) [reply docs/mkdocs/docs/technical/releasing.md:43]: Oh, you didn't create that PR initially?

**Pull request merged**